### ISSUE TITLE
RFC | Provide suggestions order RegExp

### ISSUE THREAD

**Issue by SilvaQ** [streetsidesoftware/vscode-spell-checker#1663]
# Issue 

One of the features of the language I'm proofreading is that it has vowels and consonants.
The suggested list for editing distance 1 with different vowels should be at the beginning。

for fake exmaple:
 i have word  `adam`, `x` is volwes  and i have suggested list `"adom", "adim", "adem", "adxm"`. the diff is o,i,e,x。
when I want to edit distance 1 and change the character to X words to be guaranteed to appear at the beginning of the suggested list, this cannot be done now.

If you control the order of the recommendation list with a regular expression, you should be able to do this。

# Code I am testing
Here is a simple test case I did：

```javascript
export function registerCompletionItemProvider() {
  vscode.languages.registerCompletionItemProvider("markdown", {
    provideCompletionItems() {
      // get the difference char from targer not in src
      const difference = (src, target) => {
        return Array.from(target).filter(
          (char) => !Array.from(src).includes(char)
        );
      };

      const orderReg = new RegExp("[x]");
      //const orderReg = new RegExp("[a-z]");

      const result = ["adom", "adim", "adem", "adxm"].map((item) => {
        const i = new vscode.CompletionItem(
          item,
          vscode.CompletionItemKind.Text
        );

        // for example my word is "a"
        if (orderReg.test(difference("adam", item).join(""))) {
          i.sortText = "a";
        }

        return i;
      });

      console.log("result", result);

      return result;
    },
    resolveCompletionItem(item) {
      return item;
    },
  });
}

```

# The effect
The list of recommendations when I enter a when rule x is specified in the regular expression：
![image](https://user-images.githubusercontent.com/46706370/148899088-032d3e26-ca9b-4fbb-beca-d72f9bbacc40.png)

recommended list if no regular expression is specified:
![image](https://user-images.githubusercontent.com/46706370/148899219-1537ccdb-abf0-4b6f-9084-5233c8839244.png)

# Recommend
I think this is very useful in some special scenarios, so I recommend adding a regular rule configuration for suggesting list sorting

**Comment by Jason3S** [comment]
You are right, the proper sorting of suggestions with the same edit distance is difficult.

I'm not understanding how a RegExp will help with that? I understand how your code example works:
- If the simple difference between the word and the suggestions contains a character matching the regex, then sort it towards the top.

Would you be more specific how you would specify the configuration accounting for multiple weights?

**Comment by SilvaQ** [comment]
For example, my language-> Uighur  has 24 consonants and 8 vowels in the alphabet.


![image](https://user-images.githubusercontent.com/46706370/148911564-b3693cd9-dbbd-4913-99d9-e446a953036c.png)

Any eight vowels in the substitution list should appear at the beginning of the suggestion list, regardless of the editing distance, regardless of the weight.

If not, they're pretty far down the list of 10,20 suggestions。

If there is a regular expression to specify the function, I just put these 8 characters in, and whenever they match they are put at the top of the list for me the efficiency of the recommendation goes up many times。

**Comment by Jason3S** [comment]
It seem that you want a weighted edit distance to be used in the sorting.

The rules are similar in English replacing an `e` with an `a` should be cheap. But, replacing a `e` with a `t` should be expensive. Inserting an accent `’` should be very cheap. Since these are common mistakes.

**Comment by SilvaQ** [comment]
Yes, just like the scene you said
